freqtrade/exchange/exchange.py

```diff
@@ -785,6 +785,26 @@ def _get_stake_amount_considering_leverage(self, stake_amount: float, leverage:
 
     # Dry-run methods
 
+    def taker_or_maker(
+        self,
+        order_reason: Literal['entry', 'exit', 'stoploss'],  # TODO: stoploss
+    ):
+        order_type = self._config['order_types'][order_reason]
+        if order_type == 'market' or order_reason == 'stoploss':
```

Comment:
I'm not sure if the stoploss assumption is correct, are freqtrade stoploss limit orders always taker orders, or sometimes maker orders? Theres no config price_side for it so I"m not sure

Reply:
Depends on the exchange (and the settings)
kraken only has stop market orders - so you know what you are (taker)

binance only has stop limit orders - but "default" settings set the limit price 1% below the stop - so it's reasonable to assume that you're a taker as well. 
users can adjust the limit value - and the market can "gap down" - so you'll really only know if you're taker or maker once the order fills.

other exchanges offer both limit and market options - so you can obviously have both.

In reality (independent of stoploss) you'll know this once the order fills (also a limit order can be using taker fees).

Maybe the following assumption provides a better result:
If the order fills right when it's created - it's taker fees
If it "stays there" for at least one cycle - it's maker fees.

Reply:
> binance only has stop limit orders 

binance futures can do both

Reply:
> Maybe the following assumption provides a better / more realistic result:
> If the order fills right when it's created - it's taker fees
> If it "stays there" for at least one cycle - it's maker fees.

how would I check if a cycle has been passed?

Reply:
based on the call location (where the "close limit order" is called from).
There will only be one location where the order is created - just mark that with a "first" parameter (or whatever) - and if it fills there, it's taker, otherwise it's maker fees.

Reply:
> That's not the point - it's more that there can be all sorts of such things

I was just letting you know